### PR TITLE
Fix: skip PSR-18 tests when PHP < 7

### LINKED ISSUE #54: Failed tests with PHP 5.6
Build OK.

```
TEST 8/43 [tests/PsrHttpClientClientExceptionInterface.phpt]
========DIFF========
001+ Warning: class_implements(): Class Psr\Http\Client\ClientExceptionInterface does not exist in /builddir/build/BUILD/php56-php-pecl-psr-0.6.0/NTS/tests/PsrHttpClientClientExceptionInterface.php on line 3
002+ bool(false)
003+ 
004+ Fatal error: Interface 'Psr\Http\Client\ClientExceptionInterface' not found in /builddir/build/BUILD/php56-php-pecl-psr-0.6.0/NTS/tests/PsrHttpClientClientExceptionInterface.php on line 4
001- array(1) {
002-   ["Throwable"]=>
003-   string(9) "Throwable"
004- }
005- bool(true)
006- bool(true)
007- bool(true)
008- string(4) "test"
========DONE========
FAIL Psr\Http\Client\ClientExceptionInterface [tests/PsrHttpClientClientExceptionInterface.phpt] 

TEST 9/43 [tests/PsrHttpClientClientInterface.phpt]
========DIFF========
001+ Parse error: syntax error, unexpected ':', expecting ';' or '{' in /builddir/build/BUILD/php56-php-pecl-psr-0.6.0/NTS/tests/SampleClient.inc on line 9
001- bool(true)
002- array(1) {
003-   ["Psr\Http\Client\ClientInterface"]=>
004-   string(31) "Psr\Http\Client\ClientInterface"
005- }
006- string(25) "SampleClient::sendRequest"
007- object(SampleRequest)#%d (0) {
008- }
009- bool(true)
========DONE========
FAIL Psr\Http\Client\ClientInterface [tests/PsrHttpClientClientInterface.phpt] 

TEST 10/43 [tests/PsrHttpClientNetworkExceptionInterface.phpt]
========DIFF========
001+ Parse error: syntax error, unexpected ':', expecting ';' or '{' in /builddir/build/BUILD/php56-php-pecl-psr-0.6.0/NTS/tests/PsrHttpClientNetworkExceptionInterface.php on line 8
001- array(2) {
002-   ["Psr\Http\Client\ClientExceptionInterface"]=>
003-   string(40) "Psr\Http\Client\ClientExceptionInterface"
004-   ["Throwable"]=>
005-   string(9) "Throwable"
006- }
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011- string(4) "test"
012- string(23) "MyException::getRequest"
013- object(SampleRequest)#%d (0) {
014- }
========DONE========
FAIL Psr\Http\Client\NetworkExceptionInterface [tests/PsrHttpClientNetworkExceptionInterface.phpt] 

TEST 11/43 [tests/PsrHttpClientRequestExceptionInterface.phpt]
========DIFF========
001+ Parse error: syntax error, unexpected ':', expecting ';' or '{' in /builddir/build/BUILD/php56-php-pecl-psr-0.6.0/NTS/tests/PsrHttpClientRequestExceptionInterface.php on line 8
001- array(2) {
002-   ["Psr\Http\Client\ClientExceptionInterface"]=>
003-   string(40) "Psr\Http\Client\ClientExceptionInterface"
004-   ["Throwable"]=>
005-   string(9) "Throwable"
006- }
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011- string(4) "test"
012- string(23) "MyException::getRequest"
013- object(SampleRequest)#%d (0) {
014- }
========DONE========
FAIL Psr\Http\Client\RequestExceptionInterface [tests/PsrHttpClientRequestExceptionInterface.phpt] 

```

### PR DESCRIPTION
**Pull request by flyinghail**
fix #54

### CODE REVIEW

**Comment by remicollet** [comment]
fixes issue #54 

BTW? will be better to have reason why skipped
`    die("skip xxxxxx");`

**Comment by flyinghail** [comment]
@remicollet 
PSR-18 http client interfaces requires PHP > 7.
You can check PSR Http Client's [composer.json](https://github.com/php-fig/http-client/blob/master/composer.json) and [psr/http-client](https://packagist.org/packages/psr/http-client)

**Comment by remicollet** [comment]
Sorry I don't think the PSR-18 requires any PHP version.
Only its implementation.
BTW, not a big issue.

**Comment by sergeyklay** [comment]
Actually I agree with @remicollet. Theoretically, we could implement PSR-18 even using PHP 5.5.

File: tests/PsrHttpClientClientExceptionInterface.phpt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 --TEST--
 Psr\Http\Client\ClientExceptionInterface
 --SKIPIF--
-<?php if( !extension_loaded('psr') ) die('skip '); ?>
+<?php if( !extension_loaded('psr') || PHP_MAJOR_VERSION < 7 ) die('skip '); ?>
```
Comment (sergeyklay) [inline tests/PsrHttpClientClientExceptionInterface.phpt]:
```suggestion
<?php include('skip_for_php5.inc'); ?>
```

```php
<?php
// skip_for_php5.inc

if (!extension_loaded("psr")) {
    print "skip The psr extension is not loaded";
} else if (PHP_MAJOR_VERSION < 7) {
    print "skip Due to version incompatibility";
}
?>
```

@flyinghail What do you think about something like this? Also as I know the `die` is not necessary. The `skip` word is enough.

**Comment by flyinghail** [comment]
PSR-18 uses [return type declarations](http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration), a new feature added since PHP7.

PSR-15, PSR-17 are also similar.

File: tests/PsrHttpClientClientExceptionInterface.phpt (hunk 1 above)
Comment (flyinghail) [reply tests/PsrHttpClientClientExceptionInterface.phpt]: I think it's a good idea. 

Maybe we can use similar codes to support all test files?
For example:

```php
<?php
// skip.inc

if (!extension_loaded('psr')) {
    die('skip The psr extension is not loaded');
}
```
```php
<?php
// skip_for_php5.inc

include('skip.inc');
if (PHP_MAJOR_VERSION < 7) {
    print 'skip Due to version incompatibility';
}
```

```php
<?php
// skip_for_php7.inc

include('skip.inc');
if (PHP_MAJOR_VERSION >= 7) {
    print 'skip Due to version incompatibility';
}
```

Comment (sergeyklay) [reply tests/PsrHttpClientClientExceptionInterface.phpt]: @flyinghail Looks good to me. Could you change current tests?

**Pull request merged**